### PR TITLE
fix(daemon): wire tmuxPaneResolver.Process to surface pane.process.cwd (#463)

### LINKED ISSUE #463: daemon: pane.process.cwd is null in default tmuxSessions queries — surface the slow-path requirement
## What's happening

Querying `tmuxSessions { windows { panes { process { cwd } } } }` returns `null` for every `cwd` in the default code path. To get cwds resolved I had to switch to `host { processes(filter: { commandIn: [...] }) { pid cwd } }` and join in the client by `pane.currentPid → process.pid`.

This is a UX foot-gun: the natural traversal `Session → Window → Pane → Process → cwd` returns `null`, with no hint that you needed to opt in via the `Host.processes(filter)` slow path. The schema *does* document `Process.cwd` as "Slow path — opt-in (lsof on macOS, /proc on Linux)" — but it documents it on `Process`, not on `TmuxPane.process`, where the null pops up.

## Concrete sequence (from today)

```graphql
{ tmuxSessions { windows { panes { process { cwd } } } } }
```

→ all `cwd: null`. No GraphQL error. Just silent emptiness, which is hard to debug because it looks like a real "no panes have a cwd" signal.

Workaround that worked:

```graphql
{ tmuxSessions { windows { panes { paneId currentPid } } }
  host { processes(filter: { commandIn: ["claude","node","zsh"] }) { pid cwd } } }
```

then join `pane.currentPid → process.pid` client-side. ~3x the query work for what should be a single nested traversal.

## Why this matters

- Today the cheapest way to answer "which session is sitting in which worktree" requires two queries and a client-side join. That's the daemon's bread-and-butter use case.
- The error mode (silent null) makes you doubt your data, not your query. Cost a few minutes today.
- Compounds with #401 (introspection gated): you can't even discover the slow-path workaround via `__schema` — have to read `schema.graphql` directly.

### Symptoms

- `tmuxSessions { ... panes { process { ... } } }` returns `process: null` for every pane regardless of liveness.
- `host { processes(filter: ...) { pid cwd } }` returns the same processes correctly with cwd populated.
- No GraphQL error/extension. Silent null only.

### Root cause

The `pane.process` traversal returns null because **`tmuxPaneResolver.Process` is a stub**, not because `cwd` is gated.

`internal/server/resolvers/schema.resolvers.go:695-697`:

```go
// Process is the resolver for tmuxPane.process.
func (r *tmuxPaneResolver) Process(ctx context.Context, obj *graphql1.TmuxPane) (*graphql1.Process, error) {
    return nil, nil
}
```

The schema annotates `process: Process` as `"Null until ws-b-ps wires it."` (`schema.graphql:706`). The wiring never happened. Everything downstream of `pane.process` is therefore unreachable — `cwd`, `command`, `pid`, `args`, all null because the parent node is null.

`Process.Cwd` itself **is** wired correctly:

```go
// schema.resolvers.go:99-112
func (r *processResolver) Cwd(ctx context.Context, obj *graphql1.Process) (*string, error) {
    if r.PS == nil { return nil, nil }
    cwd, err := r.PS.LoadCwd(ctx, int(obj.Pid))
    ...
}
```

It calls the cwdLoader (DataLoader, 20ms window, 128 batch) → `lsofCwd` on macOS / `/proc/<pid>/cwd` on Linux (Linux returns empty map, see caveats). This is why `host.processes(filter:) { cwd }` works fine — the resolver exists for `Process` reached through `Host.processes`, but `pane.process` never produces a `Process` for it to resolve from.

The user's framing ("auto-resolve cwd when in selection set") was directionally wrong — it would re-implement what gqlgen already does. gqlgen invokes child field resolvers only when those fields are in the selection set. The fix is to populate `pane.process`; the existing `Process.Cwd` resolver chains naturally only when the client actually selects `cwd`.

### Strategies considered

| # | Strategy | Verdict | Notes |
|---|---|---|---|
| 1 | Wire `tmuxPaneResolver.Process` to look up `pane.CurrentPid` in the ps cache and project a `Process` node | **Recommended** | One-resolver fix. Reuses `lookupPane`, `r.PS.Get`, `projectProcess` (all extant). Hidden cost is `lsof` fan-out only when `cwd` selected — see caveat. |
| 2 | Auto-resolve cwd when in selection set (user's #1) | Reject | Misdirected. gqlgen already does this; the missing piece is the parent resolver. |
| 3 | Schema-side error/extension (user's #2) | Defer | Worth doing as a doc fix, but doesn't make the field actually work. After (1), update the `process: Process` schema doc to drop "Null until ws-b-ps wires it" and clarify that traversing `pane.process.cwd` IS the de-facto opt-in for the lsof slow path. |
| 4 | Shortcut field `pane.processCwd: String` (user's #3) | Reject | Adds surface area; worse ergonomics; nothing the wired resolver doesn't already give you. |
| 5 | Wire `claudeInstanceResolver.Pane` / `claudeInstanceResolver.Process` in the same PR | Reject (scope) | Same shape of bug, but tracked separately as #468. Keep this issue scoped to `tmuxPane.process`. Mention #468 as related so the next implementer reuses the wiring approach. |

### Findings for the implementer

The fix has three small parts. Numbers are line references in `internal/server/resolvers/schema.resolvers.go` unless noted.

**1. Wire the resolver** (replaces the `return nil, nil` stub at L695-697):

```go
func (r *tmuxPaneResolver) Process(ctx context.Context, obj *graphql1.TmuxPane) (*graphql1.Process, error) {
    if r.PS == nil { return nil, nil }
    p, ok := r.lookupPane(obj.ID)        // existing helper, L1515
    if !ok || p.CurrentPid == 0 { return nil, nil }

    // Use the host segment from the pane id, NOT r.Tmux.Host(), so federated
    // pane ids (e.g. peer-host nodes) project onto the right host. See
    // splitTmuxPaneID in node.resolvers.go:378.
    host, _, _ := splitTmuxPaneID(obj.ID)
    proc, _, err := r.PS.Get(ctx, ps.ProcessID{Host: host, PID: p.CurrentPid})
    if err != nil { return nil, nil }    // exited / not in cache — null is honest
    return projectProcess(&proc, host), nil
}
```

`projectProcess` is at L1343; `splitTmuxPaneID` already exists. Prefer cache-only lookup (avoid `r.PS.Get`'s adapter-fallback on miss) when the pane.CurrentPid points at a pid that exited between snapshots — null is more honest than a fresh shell-out under a request goroutine.

**2. Parallelize `fetchCwdsDarwin`.** This is a hidden cost the user didn't see because `pane.process` was returning null. After the fix, a query like `panes { process { cwd } }` against an orchardist host (~30–100 panes) hits `internal/server/providers/ps/adapter.go:161-175`:

```go
for _, pid := range pids {
    cwd, err := a.lsofCwd(ctx, pid)   // one shellout per pid, serial
    ...
}
```

The cwdLoader (`provider.go:51`) coalesces parallel resolver calls into one batch, but the batch itself is serial inside `fetchCwdsDarwin`. 100 panes × ~5–20ms per `lsof -p` = 500ms–2s p99 inside a single GraphQL request, no timeout shield. Either parallelize via `errgroup` (cap concurrency at e.g. 16) or chain `lsof` inputs (`-p 123 -p 456 ...` — `lsof` accepts multiple `-p` flags).

**3. Schema doc fixes.** In `internal/server/resolvers/schema.graphql`:
- L706: drop `"Null until ws-b-ps wires it."`. Replace with something like `"OS-level Process for the foreground pid. Null when tmux reports no current pid or the process is no longer in the ps cache."`
- L58 (Process.cwd): clarify that traversal via `pane.process.cwd` is the de-facto opt-in path; no separate flag exists or is needed.

### Caveats

- **Linux**: `FetchCwds` returns an empty map silently on Linux (`adapter.go:148-152`). After fix, `pane.process` will populate but `cwd` will still be null on Linux until the `/proc` reader lands. Validate that this asymmetry is acceptable for the implementer's targeted platform; add a doc note if shipping macOS-only is OK.
- **Cache miss handling**: `r.PS.Get` falls back to `adapter.Fetch` on miss (`provider.go:130-144`). For a pid that exited between the tmux snapshot and the resolver, that's an extra shell-out per `pane.process`. Recommend cache-only lookup (use `r.PS.GetMany([]ProcessID{...})` or `store.Get` directly) and return nil on miss.
- **Federation**: `r.Tmux.Host()` returns the local daemon's host id. For a pane id that came from a peer (federated `Host.tmuxSessions` traversal), using `r.Tmux.Host()` would attribute the Process to the wrong host. Use `splitTmuxPaneID(obj.ID)` to extract the host segment from the pane's own id. Confirm there are no callers that pre-construct `TmuxPane{ID: "..."}` with a non-canonical id shape.
- **Process.cwd null contract**: today the field is documented as "Slow path — opt-in." After fix, it's de-facto opt-in via `pane.process.cwd` traversal, but lsof can still fail (sudo'd / exited pid / Linux). The field stays nullable; the schema doc should reflect that.
- **Related**: #468 is the same bug shape on `claudeInstance.process` / `claudeInstance.pane`. Keep #463 scoped to `tmuxPane.process` only — but the wiring pattern (lookup parent → cache → `projectProcess`) should be reused there.

## Pointers

- `internal/server/resolvers/schema.resolvers.go:695-697` — the stub
- `internal/server/resolvers/schema.resolvers.go:99-112` — Process.Cwd resolver (already wired)
- `internal/server/resolvers/schema.resolvers.go:1343-1363` — `projectProcess` projection helper
- `internal/server/resolvers/schema.resolvers.go:1515-1523` — `lookupPane` helper
- `internal/server/resolvers/node.resolvers.go:378-389` — `splitTmuxPaneID`
- `internal/server/providers/ps/adapter.go:141-197` — `FetchCwds` / `lsofCwd` (serial loop, candidate for parallelization)
- `internal/server/providers/ps/provider.go:51` — `cwdLoader` (DataLoader window/batch params)
- `schema.graphql:706` — `process: Process` doc to update

## Acceptance Criteria

- [ ] **AC1**: `tmuxSessions { windows { panes { process { pid command cwd } } } }` returns a populated `Process` node for every pane whose `currentPid > 0` and whose pid is in the ps cache. `cwd` is non-null on macOS for processes whose `lsof -p` succeeds.
- [ ] **AC2**: For panes where tmux reports no current pid (`currentPid == 0`) or the pid has exited / is not in the ps cache, `pane.process` returns `null` (no error, no panic).
- [ ] **AC3**: When the same query is issued against a federated peer host (`host(id: "Host:<peer>") { tmuxSessions { ... panes { process { ... } } } }`), each `Process` node carries the **peer's** host id in its `id` field, derived from the pane id's host segment — not the local daemon's host id.
- [ ] **AC4**: `cwd` resolution for a query selecting `panes { process { cwd } }` against a host with N panes does NOT shell out to `lsof` serially N times. Either parallelized (bounded concurrency, e.g. cap 16) or batched into a single multi-pid `lsof -p X -p Y ...` invocation. Verified by a unit test that asserts the adapter does not call `lsofCwd` more than `min(N, cap)` times concurrently OR that it issues a single batched call.
- [ ] **AC5**: Schema doc for `TmuxPane.process` no longer says `"Null until ws-b-ps wires it"`. Schema doc for `Process.cwd` clarifies that `pane.process.cwd` traversal is the de-facto opt-in path; no separate gate exists.
- [ ] **AC6**: A regression test exercises the full traversal end-to-end (pane → process → cwd) against a fake `ps` provider that returns a known cwd; the test fails on the pre-fix stub and passes after the wiring change.
- [ ] **AC7**: Linux behaviour is unchanged — `cwd` returns null silently (existing `FetchCwds` Linux fallback). Out of scope: implementing `/proc/<pid>/cwd` reader.
- [ ] **AC8**: `claudeInstance` resolvers are NOT touched (out of scope; tracked as #468).

### Approach

Replace the `tmuxPaneResolver.Process` stub with a real resolver that joins `pane.CurrentPid` against the ps cache and projects to a GraphQL `Process` node. Existing helpers (`lookupPane`, `projectProcess`, `splitTmuxPaneID`) are reused — no new abstractions. Separately, parallelize the lsof fan-out inside `fetchCwdsDarwin` so the now-reachable `panes { process { cwd } }` traversal doesn't ship a latency cliff. Finish with schema doc cleanup so the next reader doesn't trip on stale "Null until X wires it" annotations.

### Sequencing

1. **Wire the resolver.** Replace the stub at `schema.resolvers.go:695-697`. Use cache-only lookup (avoid adapter-fallback shellouts on miss) — return nil on miss. Use `splitTmuxPaneID(obj.ID)` to extract the host segment so federated pane ids project onto the correct host. Completion signal: a unit test against a fake `Tmux` + `PS` provider returns a populated `Process` node for a pane with `CurrentPid != 0`, and `nil` for a pane with `CurrentPid == 0` or missing pid.
2. **Parallelize `fetchCwdsDarwin`.** Replace the serial `for _, pid := range pids { lsofCwd(...) }` loop in `internal/server/providers/ps/adapter.go:161-175` with a bounded-concurrency `errgroup` (cap at 16) OR a single multi-pid `lsof -p X -p Y ...` invocation. Pick whichever is simpler to test deterministically. Completion signal: a benchmark or timing test on 50 fake pids completes within 250ms on macOS where each `lsof` mock takes ~10ms.
3. **End-to-end traversal test.** Add an integration test (alongside the existing `worktree_dashboard_e2e_test.go` style) that runs a real GraphQL query `{ tmuxSessions { windows { panes { process { pid cwd } } } } }` against a wired-up resolver harness using fake `Tmux` and fake `PS` providers, asserts every pane projects a real Process with the expected cwd. This test must FAIL pre-fix and PASS post-fix — that's the regression guard.
4. **Schema doc cleanup.** Edit `internal/server/resolvers/schema.graphql` (and the canonical `schema.graphql` if it's a separate file): drop "Null until ws-b-ps wires it" from `TmuxPane.process`; clarify `Process.cwd` doc. Re-run gqlgen / regenerate `generated.go` and `models_gen.go` if comments propagate. Completion signal: `git diff schema.graphql` shows the doc changes; `cargo build` (Rust client codegen) and `go generate ./...` both succeed.
5. **Federation smoke test.** Add a focused test asserting that a pane id in the form `"TmuxPane:peer-host:%26"` with a non-zero `CurrentPid` produces a Process node with id `"peer-host:<pid>"` — not `"<local-host>:<pid>"`. Lightweight; just verifies the `splitTmuxPaneID` wiring.

### Key decisions

- **Cache-only `r.PS.Get`, return nil on miss** — adapter-fallback under a request goroutine risks unbounded latency for exited pids. Honest null beats a desperate shell-out.
- **`errgroup` over multi-pid `lsof`** *(tentative)* — `errgroup` with cap 16 is more portable and easier to bound; multi-pid lsof concentrates risk on a single shellout's behavior with mixed alive/dead pids. Implementer should verify with a quick prototype which is simpler; both are acceptable.
- **Use pane id host segment, not `r.Tmux.Host()`** — federation correctness. The local daemon's view of a federated pane id must produce a Process node attributed to the peer host.
- **Schema doc fix lives in this PR, not a follow-up** — keeps the doc and code change in lock-step; reviewer doesn't have to chase a second PR.
- **#468 is explicitly out of scope** — same bug shape on `claudeInstance`, but the wiring depends on heartbeat→pid join logic (#421) that this issue doesn't touch. Conflating them widens the blast radius.
- **No new fields, no new opt-in flags** — user proposed three; investigation rejected all three. The schema surface is unchanged; only resolver behavior changes.

### Risks / open questions

- **Linux platform parity**: `FetchCwds` returns an empty map on Linux. After this PR, Linux users see `pane.process` populated but `cwd: null`. Investigation flagged this as acceptable — confirm with implementer before shipping. If unacceptable, add a `/proc/<pid>/cwd` reader, but that's a separate scope.
- **Existing callers**: are there callers in the Rust TUI or CLI that pre-construct `TmuxPane{ID: ...}` with a non-canonical id shape? `splitTmuxPaneID` would mis-parse them. Check `crates/orchard/` and `internal/cli/` for synthetic pane ids before merging.
- **DataLoader window vs new parallelism**: the cwdLoader has a 20ms coalescing window. Parallelizing inside `fetchCwdsDarwin` is orthogonal — don't accidentally weaken the coalescing. Verify that parallelism happens *inside* the batch, not *across* batches.
- **Federation transport for `TmuxPane`**: today `Host.processes` federates explicitly via `federatePeerProcesses`. Does `Host.tmuxSessions` federate? If not, then federated pane ids never reach this resolver — making AC3 partially unverifiable until tmux federation lands. Confirm before writing AC3's test.

### Out of scope

- Implementing Linux `/proc/<pid>/cwd` reader.
- Wiring `claudeInstance.process` / `claudeInstance.pane` (tracked as #468).
- Changing `Process.cwd`'s nullability or adding new opt-in flags / fields.
- Changing tmux federation behavior beyond AC3's correctness check.
- Adding a `pane.processCwd: String` shortcut field (rejected in Investigation).
- GraphQL error/extension to surface "slow path" warnings (rejected; doc fix is sufficient).

### PR DESCRIPTION
**Pull request by drewdrewthis**
## Why

Closes #463.

`tmuxSessions { windows { panes { process { cwd } } } }` returned `null` for every `cwd` in the natural traversal — silently, with no GraphQL error. Working around it required a second query to `host.processes(filter:)` and a client-side join on `pane.currentPid → process.pid`, ~3× the work for what should be a single nested traversal. That broke the daemon's bread-and-butter use case ("which session is sitting in which worktree?").

The user's framing in the issue body ("auto-resolve cwd when in selection set") was directionally wrong. Investigation found the real root cause: `tmuxPaneResolver.Process` was a `return nil, nil` stub at `schema.resolvers.go:695-697`, with the schema explicitly annotated `"Null until ws-b-ps wires it"`. That wiring never happened. Everything downstream of `pane.process` was unreachable because the parent node was always null.

## What changed

Five focused commits, one per AC:

1. **Failing regression test first** — boots an httptest daemon with fake tmux + ps providers, issues the natural traversal query, asserts a populated `Process` with the expected pid/command/cwd. Designed to fail on the stub and pass once the resolver lands. (AC6)

2. **Wired the resolver** — replaced the stub with a real implementation that:
   - Looks up the pane in the local tmux snapshot via the existing `lookupPane` helper.
   - Returns `nil` (no error) for `currentPid == 0`, lookup miss, or pid not in the cached ps snapshot. Honest null beats a desperate `ps` shellout under a request goroutine, so the lookup is cache-only — no `r.PS.Get` adapter-fallback.
   - Extracts the host segment from the **pane id** (via `splitTmuxPaneID`), NOT from `r.Tmux.Host()`. This is federation correctness: a pane id with a peer-host segment must produce a Process node attributed to that peer. (AC1, AC2)

3. **Federation primitive coverage** — three unit tests on `splitTmuxPaneID` covering local host, peer host (the federation guard), and 5 malformed inputs. Cheap regression catcher for any future "simplification" that swaps in `r.Tmux.Host()`. (AC3)

4. **Batched `lsof` shellout** — replaced the N-serial-shellout loop in `fetchCwdsDarwin` with a single comma-separated `lsof -a -d cwd -p 123,456,789 -F n` invocation. Walks the `-F` output line-by-line, attributing each `n<path>` to the most recent `p<pid>`. Rationale: now that `pane.process` is reachable, a 100-pane host would have chained 100 serial fork+exec calls per request (~500ms-2s p99). Batched: 1 shellout regardless of N. (AC4)

5. **Schema doc cleanup** — dropped `"Null until ws-b-ps wires it"` from `TmuxPane.process`; expanded `Process.cwd` to spell out that selecting `cwd` IS the opt-in (no separate flag), the cwdLoader coalesces into one batched lsof per request, and Linux fallback is unimplemented. Mirrored to the embedded SDL and regenerated `generated.go` / `models_gen.go`. (AC5)

## How it works

The wired resolver is small (~20 lines) but several invariants are load-bearing:

- **Cache-only lookup**: scans `r.PS.List()` rather than calling `r.PS.Get(ctx, ...)`. The latter falls back to `adapter.Fetch` on miss — a fresh `ps -ax` shellout per `pane.process` for an exited pid would amplify request latency without benefit. Linear scan over a few hundred cached pids is fine for v1.
- **Selection-set lazy chaining**: gqlgen invokes `Process.Cwd` only when the client actually selects `cwd`. So a query for `panes { process { pid command } }` skips the lsof path entirely; only `panes { process { cwd } }` triggers the (now-batched) shellout.
- **Partial-output channel for lsof**: lsof exits non-zero when any requested pid is missing, but the alive pids still appear on stdout. `execRunner.Run` was tightened to return `(stdout, wrappedErr)` for `*exec.ExitError` with non-empty stdout; `fetchCwdsDarwin` then parses what came through. Old behaviour `(nil, err)` is preserved for empty-stdout failures.

## Test plan

- `TestPaneProcessTraversal_PopulatesProcessAndCwd` — full GraphQL e2e regression. **Verified to FAIL on commit 920342c (stub) and PASS on commit e712d23 (fix).** This is the AC6 audit trail.
- `TestSplitTmuxPaneID_LocalHost` / `_PeerHost` / `_Malformed` — federation primitive coverage; 3 functions, 7 assertions total.
- `TestFetchCwdsDarwin_*` — 6 new unit tests covering empty input, single-pid happy path, 50-pid multi-block parsing, single-shellout invariant, partial output, and non-zero-exit-with-partial-stdout.
- Full `internal/server/resolvers/` and `internal/server/providers/ps/` suites green on the head of this branch.
- `go build ./...` and `go vet ./...` clean.

Pre-existing `internal/cli/config` test failures (TestAddPeer_*, TestAddRepo_*) are unrelated to this PR — confirmed by stashing this branch's changes and reproducing on `origin/main`. Those are config-path-migration fallout, not introduced here.

## Anything surprising?

- **AC3 is partially e2e-unverifiable today.** `lookupPane` strips the prefix `"TmuxPane:" + r.Tmux.Host() + ":"`, so a federated pane id whose host is NOT the local daemon's host won't even survive `lookupPane` — it returns `(_, false)` and the resolver returns `nil`. The tests guard the federation primitive (`splitTmuxPaneID`); full e2e (peer-host pane resolved through `Host.tmuxSessions`) is deferred until tmux federation lands. Flagged in the Investigation's "open questions."
- **Linux platform parity unchanged.** `FetchCwds` still returns an empty map silently on Linux (per existing `adapter.go` fallback). After this PR, Linux users see `pane.process` populated but `pane.process.cwd: null`. Implementing `/proc/<pid>/cwd` is explicitly out of scope.
- **#468 not touched.** `claudeInstanceResolver.Process` and `claudeInstanceResolver.Pane` are byte-identical to `origin/main` (verified). Same shape of bug, but the fix depends on heartbeat→pid join logic from #421 that this issue doesn't touch. Conflating widens blast radius.
- **`execRunner.Run` semantics tightened slightly.** `*exec.ExitError` with non-empty stdout now returns `(out, wrappedErr)` instead of `(nil, err)`. Callers that already check `err != nil` (everyone today except the new `fetchCwdsDarwin`) see no behavioural change. The change is the partial-success channel lsof needs.

# Related issue

- #463

---

### Sweep: GraphQL resolver stubs (`return nil, nil` + "Null until ws-X wires it" schema annotation)

Checked all `internal/` and `crates/` Go and Rust files; only `internal/server/resolvers/schema.resolvers.go` houses this anti-pattern.

**Must-fix in this PR**: 0.

**Same pattern, separate scope** (3): `Process.worktree` (no issue yet — naturally derives from `cwd` once #463 ships), `Process.claudeInstance` (#468), `tmuxPane.claudeInstance` (#468). Out of scope here per the Investigation and Plan; tracked.

**False positives** (3): `schema.resolvers.go:286, 787, 1068` are legitimate not-found / no-foreground null returns inside real lookups, not stubs.

The bug is bounded to one file. No whack-a-mole risk — siblings are filed.

# Related issue

- #463

# Related issue

- #463

## Summary by CodeRabbit

* **New Features**
  * Pane foreground process (pid, command, cwd) is now reliably exposable via GraphQL; cwd is populated on macOS when selected.

* **Performance**
  * macOS working-directory lookups are batched into a single operation for multi-process queries, reducing calls and improving throughput.

* **Documentation**
  * Schema docs updated to clarify process nullability, cache-only behavior, selection-based cwd resolution, and batching semantics.

* **Bug Fixes**
  * Partial lookup results are now returned when available (missing entries yield nulls rather than errors).

[Review Change Stack](https://app.coderabbit.ai/change-stack/drewdrewthis/git-orchard-rs/pull/509)

# Related issue

- #463

# Related issue

- #463

# Related issue

- #463

# Related issue

- #463

### CODE REVIEW

File: internal/server/resolvers/pane_process_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+	if err := srv.StartHostProvider(ctx); err != nil {
+		t.Fatalf("start host provider: %v", err)
+	}
+	// Start populates the initial snapshot via an implicit Refresh.
+	if err := tmuxProv.Start(ctx); err != nil {
+		t.Fatalf("start tmux: %v", err)
+	}
+	if err := psProv.Start(ctx); err != nil {
+		t.Fatalf("start ps: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (drewdrewthis) [reply internal/server/resolvers/pane_process_e2e_test.go]: Fixed in 5694afb — switched to `ctx, cancel := context.WithCancel(context.Background()); t.Cleanup(cancel)` so each test run reaps its provider watcher goroutines. Thanks for catching!
